docs/source/dataset.rst

```diff
@@ -0,0 +1,232 @@
+
+Raster and vector file compatibility
+------------------------------------
+
+Rasters to be used must be in a format compatible with 
+`rasterio <https://rasterio.readthedocs.io/en/latest/quickstart.html?highlight=supported%20raster%20format#opening-a-dataset-in-reading-mode>`_ 
+and `GDAL <https://gdal.org/drivers/raster/index.html>`_ (ex.: GeoTiff). Same for the labels 
+(aka annotations) for each image must be stored as polygons in a `Geopandas <https://geopandas.org/en/stable/docs/user_guide/io.html>`_ 
+compatible vector file.
+
+.. _datasetsinglemultiband:
+
+Single-band vs multi-band imagery
+---------------------------------
+
+Remote sensing is known to deal with raster files from a wide variety of formats and flavors. 
+To provide as much 
+flexibility as possible with variable input formats for raster data, geo-deep-learning supports:
+1. Multi-band raster files, to be used as is (all bands needed, all bands is expected order)
+2. Multi-band raster files with more bands than needed (e.g. Actual is "BGRN", needed is "BGR")
+3. Multi-band raster files with bands in different order than needed (e.g. Actual is "BGR", needed is "RGB")
+4. Single-band raster files, identified with a common string pattern (see details below)
+5. Single-band raster files, identified as assets in a stac item (see details below)
+
+To support these variable inputs, geo-deep-learning expects the first column of an input csv to be in the 
+following formats.
+
+Multi-band raster files, used as is
+====================================
+
+This is the default and basic use. 
+
+.. list-table:: 
+   :header-rows: 1
+
+   * - raster path 
+     - ...
+   * - my_dir/my_multiband_geofile.tif
+     - ...
+   
+Multi-band raster files with more bands or different order than needed 
+======================================================================
+
+Same as above, but the expected order or subset of bands must be set in the 
+"*bands*" parameter of :ref:`dataset config <yamlparameters>`. For example:
+
+.. code-block:: yaml
+
+    dataset:
+        [...]
+        bands: [3, 2, 1]
+
+Here, if the original multi-band raster had "*BGR*" bands, geo-deep-learning will reorder these bands
+to "*RGB*" order. Also, band selection and reordering can be performed simultaneously. In this example,
+if the source raster contained "*BGRN*" bands, the "*N*" band would be removed and the resulting 
+raster would still be as "*RGB*".
+
+The ``bands`` parameter is set in the 
+`dataset config <https://github.com/NRCan/geo-deep-learning/blob/develop/config/dataset/test_ci_segmentation_multiclass.yaml>`_.
+
+.. note:: 
+
+    It's the user's responsibility to know which bands correspond to each index. When dealing with 
+    multi-band source imagery, Geo-Deep-Learning doesn't "*know*" which bands are present in the file 
+    and in which order. Good to know, Geo-Deep-Learning following the GDAL convention, bands are 

```

Comment:
following the GDAL convention, bands are... -> follows the GDAL convention, where bands are ...